### PR TITLE
Move hash

### PR DESCRIPTION
**Pull request by billingsley-john**
## Proposed changes

PR which moves the hash_value calculation method into shape.py which is inherited by all parametric shapes. This removes a lot of duplicate code from each parametric shape class as the hash_value calculation is the same in each case.

## Types of changes

What types of changes does your code introduce to the Paramak?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [x] New tests
- [x] Refactoring

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] Pep8 applied
- [x] Unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by billingsley-john** [comment]
This change also allows tests to be removed from each parametric shape as the method only has to be tested once as part of shape.py

**Comment by billingsley-john** [comment]
CircleCI tests are not working because i think this branch points to the incorrect docker image for running the tests

File: paramak/parametric_shapes/extruded_circle_shape.py [hunk 1]
```diff
@@ -51,6 +47,7 @@ def __init__(
         intersect=None,
         union=None,
         material_tag=None,
+        hash_value=None,
```
Comment (Shimwell) [inline paramak/parametric_shapes/extruded_circle_shape.py:50]: Wondering if it can be removed from here

```suggestion
```

**Comment by Shimwell** [comment]
> This change also allows tests to be removed from each parametric shape as the method only has to be tested once as part of shape.py

But don't we still need to test that is propagates up to the parametric shapes

**Comment by Shimwell** [comment]
> CircleCI tests are not working because i think this branch points to the incorrect docker image for running the tests

ah yes that might be my fault. I tried to change the image from opencmcworkshop/openmc_nndc to opencmcworkshop/openmc 
but I thought I had changed ti back

sorry about that

**Comment by Shimwell** [comment]
Do we have to include the hash_value in the class __init__ as this exposes it to the users. 
Perhaps it could be passed along with other **kwargs when needed and if it is not found in the **kwargs dict it can be set to none

**Comment by Shimwell** [comment]
lots of great code reduction

**Comment by billingsley-john** [comment]
> > This change also allows tests to be removed from each parametric shape as the method only has to be tested once as part of shape.py
> 
> But don't we still need to test that is propagates up to the parametric shapes @billingsley-john

Apologies, I forgot to mention that this has been tested. The parametric shape solid is only reconstructed when changes to its parameters have been made. The hash_value is only updated when .solid is called.

**Comment by billingsley-john** [comment]
> Do we have to include the hash_value in the class **init** as this exposes it to the users.
> Perhaps it could be passed along with other **kwargs when needed and if it is not found in the **kwargs dict it can be set to none

Happy to implement something like this if we think users are likely to want to specify hash_values upon construction. What do you think?

**Pull request merged**